### PR TITLE
fix: use propTypes to validate props, including tokens, in non-TS environments

### PR DESCRIPTION
**Pull request by TheSisb**
- [x] Add some proptype utils to the style-props package to make checking tokens in proptypes viable
- [x] Add missing proptypes to every package
- [x] Verified the exports on each package

### CODE REVIEW

File: packages/paste-core/components/card/src/index.tsx [hunk 1]
```diff
@@ -33,4 +34,23 @@ const Card: React.FunctionComponent<CardProps> = ({
   );
 };
 
+// TODO WIP FIXME
+const isSpaceProp = (props: {}, propName: string, componentName: string): Error | null => {
```
Comment (TheSisb) [inline packages/paste-core/components/card/src/index.tsx]: TODO use function from style-props package

File: packages/paste-core/components/card/stories/index.stories.tsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ storiesOf('Components|Card', module)
     </Card>
   ))
   .add('No Padding', () => (
-    <Card padding="space0">
+    <Card padding="2px">
```
Comment (TheSisb) [inline packages/paste-core/components/card/stories/index.stories.tsx]: TODO undo this

File: packages/paste-core/components/list/src/index.tsx [hunk 3]
```diff
@@ -51,13 +62,19 @@ const UnorderedList: React.FC<UnorderedListProps> = ({children, marginTop, margi
 UnorderedList.defaultProps = {
   marginBottom: 'space70',
 };
+UnorderedList.propTypes = {
+  marginTop: PropTypes.oneOf([]),
```
Comment (TheSisb) [inline packages/paste-core/components/list/src/index.tsx]: TODO fill all these props in with style-props functions

File: packages/paste-core/primitives/box/src/BoxPropTypes.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  flexWrap: ResponsiveProp(PropTypes.string),
+  flexDirection: ResponsiveProp(PropTypes.string),
+  flex: ResponsiveProp(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
+  flexGrow: ResponsiveProp(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
+  flexShrink: ResponsiveProp(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
+  flexBasis: ResponsiveProp(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
+  justifySelf: ResponsiveProp(PropTypes.string),
+  alignSelf: ResponsiveProp(PropTypes.string),
+  order: ResponsiveProp(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
+};
```
Comment (richbachman) [inline packages/paste-core/primitives/box/src/BoxPropTypes.ts]: Good call moving this to a separate file.

File: packages/paste-core/components/alert/src/index.tsx [hunk 5]
```diff
@@ -84,11 +84,14 @@ const Alert: React.FC<AlertProps> = ({children, onDismiss, variant, role, ...pro
...
 };
-Alert.propTypes = {
-  children: PropTypes.node.isRequired,
-  onDismiss: PropTypes.func,
-  role: PropTypes.string,
-  variant: PropTypes.oneOf(Object.keys(AlertVariants).map(variant => AlertVariants[variant])).isRequired,
-};
 Alert.displayName = 'Alert';
+
+if (process.env.NODE_ENV === 'development') {
```
Comment (SiTaggart) [inline packages/paste-core/components/alert/src/index.tsx]: Just had a thought, does this get stripped out on our production build?

Comment (TheSisb) [reply packages/paste-core/components/alert/src/index.tsx]: I had the same thought last night, and yup it does :(.  I'll just remove it for now and we should encourage folks to use the babel transform

Comment (SiTaggart) [reply packages/paste-core/components/alert/src/index.tsx]: 🎉

**Pull request merged**